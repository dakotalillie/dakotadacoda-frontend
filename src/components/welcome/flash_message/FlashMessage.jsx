import React from 'react';
import { Alert } from 'reactstrap';
import PropTypes from 'prop-types';

import './flash_message.scss';

export default function FlashMessage({ link, title, flashOpen, toggleFlash }) {
  return (
    <Alert
      color="dark"
      className="flash_message"
      isOpen={flashOpen}
      toggle={toggleFlash}
      transition={{ timeout: 0, unmountOnExit: true }}
    >
      Check out my latest blog post &mdash;{' '}
      <a href={link} className="alert-link">
        {title}
      </a>
    </Alert>
  );
}

FlashMessage.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  flashOpen: PropTypes.bool.isRequired,
  toggleFlash: PropTypes.func.isRequired,
};
