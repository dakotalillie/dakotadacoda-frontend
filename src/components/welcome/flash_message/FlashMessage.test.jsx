import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FlashMessage from './FlashMessage';

describe('FlashMessage', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FlashMessage
        title=''
        link=''
        flashOpen
        toggleFlash={() => {}}
      />
    );
  });

  it('renders successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});