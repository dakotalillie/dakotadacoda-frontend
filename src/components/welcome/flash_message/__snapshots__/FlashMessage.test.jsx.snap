// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessage renders successfully 1`] = `
<Alert
  className="flash_message"
  closeAriaLabel="Close"
  color="dark"
  isOpen={true}
  tag="div"
  toggle={[Function]}
  transition={
    Object {
      "timeout": 0,
      "unmountOnExit": true,
    }
  }
>
  Check out my latest blog post â€”
   
  <a
    className="alert-link"
    href=""
  />
</Alert>
`;
